### PR TITLE
Update Installation-Guide.rst

### PR DESCRIPTION
**Pull request by baryluk**
* Remove libnuma-dev
* Add a hint about reboot.
* Remove info about old GPG keys
* Remove irrelevant info about keyring
* Make a note about versioned ROCm, when testing it.
* Uninstall should use `--purge` as well use `*` to pick versioned packages as well, and remove configs.
* Make a note about removal, to not follow it blindly.
* Make a note about updates.
* Make notes about versioned packages and LD_LIBRARY_PATH
* Use stateful language, not "in F x changed to y", instead say: "F is y".
* Inform about archival versions repo.

### CODE REVIEW

**Comment by baryluk** [comment]
@Rmalavally

Ping. Also there is no need to constantly update this installation guide on every release of ROCm.

The update for ROCm 3.10 were pointless, and this PR does solve this too.

**Comment by Rmalavally** [comment]
Thanks for your feedback and the PR. Our QA team is validating the requested changes before we merge them into our documentation. 

AMD ROCm Documentation Team

File: Installation_Guide/Installation-Guide.rst [hunk 1]
```diff
@@ -119,7 +110,7 @@ Supported Operating Systems
 Ubuntu
 =========
 
-**Note**: AMD ROCm only supports Long Term Support (LTS) versions of Ubuntu. Versions other than LTS may work with ROCm, however, they are not officially supported. 
+**Note**: AMD ROCm only supports Long Term Support (LTS) versions of Ubuntu. Versions other than LTS may work with ROCm, however, they are not officially supported. Debian-based distributions other than Ubuntu LTS are not officially supported, they might however work. In case of issues or bugs, retest on Ubuntu LTS, if possible.
```
Comment (saadrahim) [inline Installation_Guide/Installation-Guide.rst:113]: How are you inferring this?

File: Installation_Guide/Installation-Guide.rst [hunk 2]
```diff
@@ -209,21 +195,36 @@ The current rocm.gpg.key is not available in a standard key ring distribution, b
...
+Uninstalling ROCm Packages
+'''''''''''''''''''''''''''''
 
-To uninstall the ROCm packages from Ubuntu 20.04 or Ubuntu 18.04.5, run the following command:
+To uninstall the ROCm packages from Ubuntu, run the following command:
 
 ::
 
-  sudo apt autoremove rocm-opencl rocm-dkms rocm-dev rocm-utils && sudo reboot
+  sudo apt autoremove --purge -V rocm-opencl* rocm-dkms* rocm-dev* rocm-utils* && sudo reboot
```
Comment (saadrahim) [inline Installation_Guide/Installation-Guide.rst:215]: Thank you for contributing this.

File: Installation_Guide/Installation-Guide.rst (hunk 1 above)
Comment (baryluk) [reply Installation_Guide/Installation-Guide.rst:113]: There are many people using non-LTS or non Ubuntu distros, like Debian (including me), with a success and without major issues.

**Pull request merged**